src/Nethermind/Nethermind.Merge.Plugin.Test/MergePluginTests.cs

```diff
@@ -81,6 +81,7 @@ private IContainer BuildContainer(IConfigProvider? configProvider = null)
                     api.TransactionComparerProvider!,
                     ctx.Resolve<IBlocksConfig>(),
                     api.LogManager!);
+                api.EngineRequestsTracker = NullEngineRequestsTracker.Instance;
```

Comment:
Same as in `BaseEngineModuleTests`, would a substitute be enough?

```suggestion
                api.EngineRequestsTracker = Substitute.For<IEngineRequestsTracker>();
```

Given that `NullEngineRequestsTracker` is only used in tests, if we replace all its usages with substitutes we can remove the class entirely.